packages/components/src/ClickableStyle/ClickableStyle.tsx

```diff
@@ -40,10 +68,31 @@ const ClickableStyle = React.forwardRef(
       state,
       variant,
       className,
+      icon,
+      iconProps,
       ...rest
     }: ClickableStyleProps<IComponent>,
     ref: React.ForwardedRef<HTMLElement>,
   ) => {
+    const renderIcon = (placement: IconPlacement) => {
+      const horizontalMargin = (iconProps && iconProps.horizontalMargin) || 0;
```

Comment:
Not necessary, but I think we have access to optional chaining

```suggestion
      const horizontalMargin = iconProps?.horizontalMargin ?? 0;
```